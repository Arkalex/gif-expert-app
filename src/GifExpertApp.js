import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

export const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState( defaultCategories )

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr/>

            <ol>
                {
                    categories.map( (category, i) => {
                        return <GifGrid 
                                    key={i}            
                                    category={category} 
                                />
                    })
                }
            </ol>
            {

            }
        </>
    )
}
