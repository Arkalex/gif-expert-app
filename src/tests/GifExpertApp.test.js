import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


describe('Pruebas en <GifExpertApp />', () => {
    test('should show correctly', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('should show a list of categories', () => {
        const categories = ['Dragon Ball', 'Kimetsu no Yaiba'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);


        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length);
    })
    
})