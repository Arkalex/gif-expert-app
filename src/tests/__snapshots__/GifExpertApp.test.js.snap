// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="0"
    />
    <GifGrid
      category="Kimetsu no Yaiba"
      key="1"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> should show correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
